rbx_binary/src/deserializer.rs

```diff
@@ -591,7 +752,73 @@ impl<R: Read> BinaryDeserializer<R> {
                 }
             },
             Type::Vector3 => {}
-            Type::CFrame => {}
+            Type::CFrame => match canonical_type {
+                VariantType::CFrame => {
+                    let referents = &type_info.referents;
+                    let mut rotations = Vec::with_capacity(referents.len());
+
+                    for _ in 0..referents.len() {
+                        let id = chunk.read_u8()?;
+                        if id == 0 {
+                            rotations.push(Matrix3::new(
+                                Vector3::new(
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                ),
+                                Vector3::new(
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                ),
+                                Vector3::new(
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                    chunk.read_f32::<LittleEndian>()?,
+                                ),
+                            ));
+                        } else {
+                            let special_case = special_case_to_rotation(id);
+                            if special_case.is_some() {
+                                rotations.push(special_case.unwrap());
+                            } else {
+                                return Err(InnerError::BadCFrameOrientationId {
+                                    type_name: type_info.type_name.clone(),
+                                    prop_name,
+                                    id,
+                                });
+                            }
+                        }
+                    }
+
+                    let mut x = vec![0.0; referents.len()];
+                    let mut y = vec![0.0; referents.len()];
+                    let mut z = vec![0.0; referents.len()];
+
+                    chunk.read_interleaved_f32_array(&mut x)?;
+                    chunk.read_interleaved_f32_array(&mut y)?;
+                    chunk.read_interleaved_f32_array(&mut z)?;
+
+                    for i in 0..referents.len() {
+                        let instance = self.instances_by_ref.get_mut(&referents[i]).unwrap();
+                        instance.properties.push((
+                            canonical_name.clone(),
+                            Variant::CFrame(CFrame::new(
+                                Vector3::new(x[i], y[i], z[i]),
+                                rotations[i],
```

Comment:
Can we do the iterator zipping dance like in the other implementations instead of using array indexing here? Each array access forces the compiler to check if our access is in bounds of the array, while using iterator combinators can help the compiler elide them. Because of this, numeric for loops in Rust loop are actually sort of rare!

In this particular case, I'd probably zip `x`, `y`, and `z` into a iterator that yields `Vector3`, then zip it with `referents` and `rotations`.